runtime/lib/ttnn/operations/matmul/matmul.cpp

```diff
@@ -4,20 +4,102 @@
 
 #include "matmul.h"
 #include "tt/runtime/detail/ttnn.h"
+#include "tt/runtime/detail/workarounds.h"
 #include "tt/runtime/ttnn/operations/utils.h"
+#include <optional>
 
 // ANCHOR: adding_an_op_matmul_runtime_operations
 namespace tt::runtime::ttnn::operations::matmul {
+
+// This is a workaround for the lack of program config selection in ttnn.matmul.
+// The logic here is temporary and totaly incompleate.
```

Comment:
Do we have a workaround for emitc as well?

Reply:
This is a temporary workaround just for mnist until the fix for this lands in metal. Do we run mnist trough emitc atm?

Reply:
We want to :)

In general, I don't this it's okay to implement workarounds in one of the "backends" of the IR. This puts pressure on the non-workaroundED backends to play catch up. We're already deep enough with workarounds in ttrt, adding some more will only increase the gap with emitc.

We should either:
1. workaround this on the IR level
2. wait for metal to fix and uplift
3. provide workarounds for both (all) IR "backends"

If there's a strong enough reason (e.g. it's blocking the whole team for an important deliverable, so passing around git diff patches is not sustainable), I guess we could make an exception, but let's do that only with a strong commitment and an ETA on a fix from metal team. Currently, the linked issue https://github.com/tenstorrent/tt-metal/issues/13204 doesn't seem to have an owner, nor any traction.

Reply:
As discussed offline, @skrsticTT will be taking a look at the issue on metal this week. Due to the fact that his team is in the middle of a reorg, there is no ETA nor guarantee that this will be fixed shortly.

I'm all for waiting for a fix in metal, but this was prioritized as a result of the F2F discussions, so diverting the question of how urgent this is to someone with more context @nobradovictt.

Reply:
This is not related to running MNIST or not running, this is related to running MNIST fully L1 sharded to gain/measure performance. MNIST is functionally being run already via generality path which includes emitc. Workaround is added based on instructions for runtime component to do so and properly tagged with corresponding blocking issue to remove it. Optimizer development must not be blocked on current state of other components.

Reply:
Thanks for the context @nobradovictt

> properly tagged with corresponding blocking issue to remove it

Can you tag the issue here please?

Reply:
Here it is: https://github.com/tenstorrent/tt-mlir/issues/891
```
  // TODO(bug #891): ttnn::matmul doesn't chose correct program config.
  bool setMatmul1DProgramConfig;
```

Reply:
@odjuricicTT can we add this to the workarounds env in runtime to track it there: `runtime/include/tt/runtime/detail/workarounds.h`